src/service/extensions-impl.js

```diff
@@ -328,18 +383,16 @@ export class Extensions {
   addDocFactory_(factory, opt_forName) {
     const holder = this.getCurrentExtensionHolder_(opt_forName);
     holder.docFactories.push(factory);
-  }
 
-  /**
-   * Registers a shadow-root factory.
-   * @param {function(!ShadowRoot)} factory
-   * @param {string=} opt_forName
-   * @private
-   * @restricted
-   */
-  addShadowRootFactory_(factory, opt_forName) {
-    const holder = this.getCurrentExtensionHolder_(opt_forName);
-    holder.shadowRootFactories.push(factory);
+    // If a single-doc mode, run factory right away if it's included by
+    // the doc.
```

Comment:
Single-doc auto-installs extensions, including elements and services that were declared for it. And the declarations are kept in case they are needed for other ampdocs, such as FIE.